src/XRP/issued-currency-client.ts

```diff
@@ -126,6 +136,71 @@ export default class IssuedCurrencyClient {
     return trustLines
   }
 
+  /**
+   * Returns information about the total balances issued by a given account,
+   * optionally excluding amounts held by operational addresses.
+   * @see https://xrpl.org/issuing-and-operational-addresses.html
+   *
+   * @param account The account for which to retrieve balance information, encoded as an X-Address.
+   * @param accountsToExclude (Optional) An operational address to exclude from the balances issued, or an array of such addresses,
+   *                   encoded as X-Addresses.
+   * @see https://xrpaddress.info/
+   * @returns A GatewayBalances object containing information about an account's balances.
+   */
+  public async getGatewayBalances(
+    account: string,
+    accountsToExclude?: string | Array<string>,
+  ): Promise<GatewayBalances> {
+    // check issuing account for X-Address format
+    const classicAddress = XrpUtils.decodeXAddress(account)
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+
+    // check excludable addresses for X-Address format
+    const addressesToExcludeArray: Array<string> = []
+    if (accountsToExclude !== undefined) {
+      if (typeof accountsToExclude == 'string') {
+        const excludeClassicAddress = XrpUtils.decodeXAddress(accountsToExclude)
+        if (!excludeClassicAddress) {
+          throw XrpError.xAddressRequired
+        }
+        addressesToExcludeArray.push(excludeClassicAddress.address)
+      } else {
+        for (const address of accountsToExclude) {
+          const excludeClassicAddress = XrpUtils.decodeXAddress(address)
+          if (!excludeClassicAddress) {
+            throw XrpError.xAddressRequired
+          }
+          addressesToExcludeArray.push(excludeClassicAddress.address)
+        }
+      }
+    }
```

Comment:
if `accountsToExclude` is always an array, then this can all turn into

```
accountsToExclude.forEach((account) => {
  const excludeClassicAddress = XrpUtils.decodeXAddress(accountToExclude)
  if (!excludeClassicAddress) {
    throw XrpError.xAddressRequired
  }
}
```

Reply:
So I can get rid of the check for that ugly check for a string type, but in this case I actually still have to create the array of classic addresses because this is going directly into a JSON request (as opposed to being serialized and signed) so it can't actually take X-addresses.  I'll name it a bit better